stestr/commands/run.py

```diff
@@ -127,30 +132,55 @@ def get_description(self):
         return help_str
 
     def take_action(self, parsed_args):
+        user_conf = user_config.get_user_config(self.app_args.user_config)
         filters = parsed_args.filters
         args = parsed_args
-        pretty_out = not args.no_subunit_trace
+        if getattr(user_conf, 'run', False):
+            if not user_conf.run.get('no-subunit-trace'):
+                if not args.no_subunit_trace:
+                    pretty_out = False
+                else:
+                    pretty_out = True
+            else:
+                pretty_out = False
+
+            pretty_out = args.force_subunit_trace or pretty_out
+            concurrency = args.concurrency or user_conf.run.get(
```

Comment:
I think these don't work correctly because if a user specify 0 or False for concurrency, random, color and etc. with the command line, this code just ignores them.

Probably, we need to remove the default values from add_argument() to handle here correctly?

Reply:
So I think the logic works out fine for the binary store_true flags because there isn't a way to explicitly say use the flag but as a value that evals false. The flag is either true and used or the default and false.

But you're right, with concurrency a user could manually specify 0 and we'd want that to take precedence over the user config file default. I'll have to rework the logic (and the default) a bit to handle that edge case.